Getting started guide (#15)
neo4j/cypher-builder: docs/modules/ROOT/pages/getting-started/your-first-query.adoc

= Your first query

> In this guide, you'll create a simple Cypher query to match and return Movies using the Cypher Builder:

Reviewer: ```suggestion
This tutorial shows how to start using Cypher Builder by creating a simple query to match and return Movies from the link:https://neo4j.com/docs/getting-started/appendix/example-data/[Movie database].
```

```cypher
MATCH(m:Movie)
RETURN m
```

> == Matching nodes

Reviewer: ```suggestion
```

Author: Isn't it a good idea to split the guide into separate small sections?

First, you need to define a node variable in your script. You can do it by creating a new instance of `Cypher.Node`:

```javascript
const movieNode = new Cypher.Node({
  labels: ["Movie"]
});
```

> Then, you need to create a `MATCH` clause and pass the node variable as an argument:

Reviewer: ```suggestion
.. Create a `MATCH` clause and pass the node variable as an argument:
```

```javascript
const clause = new Cypher.Match(movieNode);
```

== Returning the variable

> The second part of the query is the `RETURN` of the Node variables, you can do this with the `.return` method in the match clause:

Reviewer: In terms of style guide, we should consider whether we start writing `MATCH`/`RETURN`/etc clause or without formatting

```suggestion
. To `RETURN` the Node variables, use the `.return` method in the `MATCH` clause:
```

Author: mmm, the problem is that `MATCH` refers to the result cypher, while lowercase match clause refers to the javascript code (Bit confusing :sweat: )

```javascript
clause.return(movieNode);
```

More succinctly, this can be done when creating the clause in a single statement:

```javascript
const clause = new Cypher.Match(movieNode).return(movieNode);
```

== Building the Cypher query

> The last step, once we have our query ready, is to **build** it. This will generate the actual Cypher string that can be used in a Neo4j database:

Reviewer: ```suggestion
. After the query ready, it is time to build it. 
This step will generate the actual Cypher string that can be used in a Neo4j database:
```

```javascript
const { cypher } = clause.build();

console.log(cypher);
```

> == Conclusion

Reviewer: In case we opt for using a list

```suggestion
```

> Putting everything together, your script should look something like this:

Reviewer: ```suggestion
+
Altogether, your script should look something like this:
```
